libraries/chain/include/scorum/chain/database/budget_management_algorithms.hpp

```diff
@@ -7,6 +7,41 @@
 namespace scorum {
 namespace chain {
 
+template <typename PositionWeightList, typename PerBlockByPositionList>
+share_type calculate_vcg_cash(const size_t position,
+                              const PositionWeightList& coeffs,
+                              const PerBlockByPositionList& per_block_list)
+{
+    FC_ASSERT(coeffs.size() > 0, "Invalid coefficient's list");
+    FC_ASSERT(per_block_list.size() == coeffs.size() + 1, "Invalid list of per-block values");
+    FC_ASSERT(position < coeffs.size(), "Invalid position");
+
+    const auto top_size = coeffs.size() - 1;
```

Comment:
A little bit confusing. `top_size` should be equal `coeffs.size()`. Add `-1` to `auto inverted_position = top_size - 1;` instead